src/options.cpp

```diff
@@ -2945,48 +2968,102 @@ std::string options_manager::show( bool ingame, const bool world_options_only,
     while( true ) {
         ui_manager::redraw();
 
+        recalc_startpos = false;
         Page &page = pages_[iCurrentPage];
         auto &page_items = page.items_;
 
         auto &cOPTIONS = ( ingame || world_options_only ) && iCurrentPage == iWorldOptPage ?
                          ACTIVE_WORLD_OPTIONS : OPTIONS;
 
-        const std::string &opt_name = *page_items[iCurrentLine];
-        cOpt &current_opt = cOPTIONS[opt_name];
+        std::string action = ctxt.handle_input();
 
-        const std::string action = ctxt.handle_input();
-
-        if( world_options_only && ( action == "NEXT_TAB" || action == "PREV_TAB" ||
-                                    ( action == "QUIT" && ( !on_quit || on_quit() ) ) ) ) {
+        if( world_options_only &&
+            ( action == "NEXT_TAB" || action == "PREV_TAB" ||
+              ( ( action == "QUIT" || action == "SEC_SELECT" ) && ( !on_quit || on_quit() ) ) ) ) {
             return action;
         }
 
+        if( action == "MOUSE_MOVE" || action == "SELECT" ) {
+            bool found_opt = false;
+            sel_worldgen_tab = 1;
+            if( world_options_only ) {
+                // worldgen tabs
+                cata::optional<point> coord = ctxt.get_coordinates_text( w_options_border );
+                for( const auto &ent : worldgen_tab_map ) {
+                    if( coord.has_value() && ent.second.contains( coord.value() ) ) {
```

Comment:
You may want to put this logic for "find element by point in a std::map<..., inclusive_rectangle<point>>" into a separate function. It's repeated not only here, but also in `help::display_help` above.